src/placeos-frontend-loader/api/remote/githubremote.cr

```diff
@@ -0,0 +1,176 @@
+require "hash_file"
+require "./remote"
+
+module PlaceOS::FrontendLoader
+  class GitHubRemote < PlaceOS::FrontendLoader::Remote
+    def initialize
+    end
+
+    private alias Remote = PlaceOS::FrontendLoader::Remote
+
+    TAR_NAME = "temp.tar.gz"
+
+    # Returns the branches for a given repo
+    def branches(repo : String) : Hash(String, String)
+      uri = "https://api.github.com/repos/#{repo}/branches"
+      response = HTTP::Client.get uri
+      raise Exception.new("status_code for #{uri} was #{response.status_code}") unless (response.success? || response.status_code == 302)
+      parsed = JSON.parse(response.body).as_a
```

Comment:
`JSON.parse` is mainly for hacking stuff together/unknown data, try to migrate to using more strict JSON objects when you can